### ISSUE TITLE
Memgraph COLLECT not working properly 

### ISSUE THREAD

**Issue by antejavor** [memgraph/memgraph#1525]
**Memgraph version**
Memgraph 2.12

**Environment**
Docker 

**Describe the bug**
When running collect and distinct, the `ID` of each node is different, but properties are copied from the first node. 

**To Reproduce**
```
MATCH (n) DETACH DELETE n; 


CREATE (t:Tag {short_code: "TST", name: "TEST", description: "SYSTEM_TAG", color: "#318311"})
CREATE (t2:Tag {short_code: "PRD", name: "PROD", description: "SYSTEM_TAG", color: "#2D3EBE"})
CREATE (t3:Tag {short_code: "STG", name: "STAGE", description: "SYSTEM_TAG", color: "#8F1A1A"})

CREATE (device {name: "name1"})

MATCH (device {name: "name1"})
MATCH (t {short_code: "TST"})
MERGE (device)-[a:ASSOCIATED]-(t);

MATCH (device {name: "name1"})
MATCH (t {short_code: "PRD"})
MERGE (device)-[a:ASSOCIATED]-(t);

MATCH (device {name: "name1"})
MATCH (t {short_code: "STG"})
MERGE (device)-[a:ASSOCIATED]-(t);

MATCH (d {name: "name1"})
OPTIONAL MATCH (d)-[t:ASSOCIATED]-(tag:Tag)
RETURN {
    d: d,
    tags: collect(DISTINCT {id: ID(t), short_code: tag.short_code, description: tag.description})
}

```

Output: 
```

{
   "d": {
      "id": 30366,
      "labels": [],
      "properties": {
         "name": "name1"
      },
      "type": "node"
   },
   "tags": [
      {
         "description": "SYSTEM_TAG",
         "id": 26639,
         "short_code": "TST"
      },
      {
         "description": "SYSTEM_TAG",
         "id": 26640,
         "short_code": "TST"
      },
      {
         "description": "SYSTEM_TAG",
         "id": 26641,
         "short_code": "TST"
      }
   ]
}
--



```

**Expected behavior**
Different properties for different nodes. 

These are quite common operations, that a lot of users are using + sound like bug that should not be that hard to solve.

**Comment by antejavor** [comment]
I have updated the severity level here, because IMO there is no easy workaround for this.

**Comment by antejavor** [comment]
@Josipmrden and @as51340, this is fixed and will be going out in 2.13.?

**Comment by Josipmrden** [comment]
Yes, this is a duplicate of: https://github.com/memgraph/memgraph/issues/1517
